go.mod

```diff
@@ -1,3 +1,9 @@
 module github.com/broadinstitute/sherlock
 
 go 1.16
+
+require (
+	github.com/golang-migrate/migrate/v4 v4.14.1
```

Comment:
I'm not sure I agree that this is the right approach to take for auditing these `go.mod` files for unused dependencies. 
go.mod and the associated go.sum are both programmatically generated and only manipulated only through programmatic means as well. 

The developer essentially never touches these files directly so I'm hesitant to introduce an expectation that they should then add a comment for every entry in a generated file. Particularly since [they can get quite long ](https://github.com/kubernetes/kubernetes/blob/master/go.mod).

Also go ships with built in tooling for detecting unused dependencies. `go mod tidy` will automatically prune unreferenced deps from this file. I think I'd prefer a solution where we just run a CI check which audits for unused dependencies and fails if it finds any. The golangci-lint tool that already runs in [this CI workflow](https://github.com/broadinstitute/sherlock/blob/main/.github/workflows/lint.yaml) already checks for a huge variety of potential issues and I believe this is one of them, but I will verify. But even if it doesn't this is definitely something that we can implement an automated CI check for and I think that will be a more maintainable solution to this issue than trying to add potentially hundreds of comments to an otherwise codegen'ed file.

Reply:
everything there sounds about right. Sorry I didn't realize this was autogenerated and I guess is basically a lock file. You're totally right that touching this file in this case is a big no/no and a terrible idea.